src/sbml/SBMLTransforms.cpp

```diff
@@ -652,12 +652,35 @@ SBMLTransforms::evaluateASTNode(const ASTNode * node, const IdValueMap& values,
   case AST_FUNCTION:
     /* shouldnt get here */
     // but we do if math we are expanding uses a functionDefinition
-    lfd = m->getListOfFunctionDefinitions();
-    tempNode = node->deepCopy();
-    replaceFD(tempNode, lfd);
-    result = evaluateASTNode(tempNode, values, m);
-    delete tempNode;
-    break;
+    {
+        if(m != NULL){
+            lfd = m->getListOfFunctionDefinitions();
+        }
+        unsigned nFunctionDefinitions = 0;
+        if (lfd != NULL){
+            nFunctionDefinitions = lfd->size();
+        }
+        bool functionDefinitionExists{false};
```

Comment:
here i think it would be cleaner to just use the `get(sid)` overload of the list of functiondefinitions, something like 


     bool functionDefinitionExists = lfd->get(node->getName()) != NULL;


would that work for you? (in general usually all libsbml getters exist once for index, and once for id).

Reply:
yes that's much cleaner - thanks, I'll update the PR